app/src/main/java/ch/epfl/cs311/wanderwave/model/remote/TrackConnection.kt

```diff
@@ -77,19 +77,16 @@ class TrackConnection(private val database: FirebaseFirestore) :
 
         trackRef?.addSnapshotListener { trackDocument, error ->
           val track = trackDocument?.let { Track.from(it) }
-          profileRef?.addSnapshotListener { profileDocument, error ->
-            val profile = profileDocument?.let { Profile.from(it) }
-            if (track == null) {
-              trySend(
-                  Result.failure(Exception("Error fetching the track, firebase format is wrong")))
-            } else {
+          if (track == null) {

```

Comment:
This will be modified in another branch, can you tell me what you wanted to do so I can solve the merge conflicts?